provisioning_templates/snippet/kickstart_networking_setup.erb

```diff
@@ -1,195 +1,45 @@
-<% end -%>
-ONBOOT=yes
-PEERDNS=no
-PEERROUTES=no
-DEFROUTE=no
-<% if vlan -%>
-VLAN=yes
-<% elsif alias_type -%>
-TYPE=Alias
-<% end -%>
-NM_CONTROLLED=no
-MASTER=<%= bond.identifier %>
-SLAVE=yes
-EOF
-
-<% bonded_interfaces.push(interface.identifier) -%>
-<% end %>
-<% end %>
-
-<% @host.managed_interfaces.each do |interface| %>
-<% next if !interface.managed? || interface.subnet.nil? -%>
-<% next if bonded_interfaces.include?(interface.identifier) -%>
-
-<% subnet = interface.subnet -%>
-<% subnet6 = interface.subnet6 -%>
-<% virtual = interface.virtual? -%>
-<% vlan = virtual && (subnet.has_vlanid? || interface.vlanid.present?) -%>
-<% alias_type = virtual && !subnet.has_vlanid? && interface.identifier.include?(':') -%>
-<% dhcp = subnet.dhcp_boot_mode? -%>
-
-# <%= interface.identifier %> interface
-real=`ip -o link | grep <%= interface.inheriting_mac -%> | awk '{print $2;}' | sed s/:$//`
-<% if virtual -%>
-  real=`echo <%= interface.identifier -%> | sed s/<%= interface.attached_to -%>/$real/`
-<% end -%>
-
-# ifcfg files are ignored by NM if their name contains colons so we convert colons to underscore
-<% if alias_type -%>
-sanitized_real=`echo $real | sed s/:/_/`
-<% else -%>
-sanitized_real=$real
-<% end -%>
-
-cat << EOF > /etc/sysconfig/network-scripts/ifcfg-$sanitized_real
-BOOTPROTO="<%= dhcp ? 'dhcp' : 'none' -%>"
-<% unless dhcp -%>
-IPADDR="<%= interface.ip -%>"
-NETMASK="<%= subnet.mask -%>"
-<% if subnet && subnet.gateway.present? -%>
-GATEWAY="<%= subnet.gateway %>"
-<% end -%>
-<% end -%>
-DEVICE=$real
-<% unless virtual -%>
-HWADDR="<%= interface.mac -%>"
-<% end -%>
-ONBOOT=yes
-<% if interface.respond_to?(:primary) && interface.primary -%>
-PEERDNS=yes
-PEERROUTES=yes
-DEFROUTE=yes
-<% if !dhcp && subnet.dns_primary.present? -%>
-DNS1=<%= subnet.dns_primary %>
-<% if subnet.dns_secondary.present? -%>
-DNS2=<%= subnet.dns_secondary %>
-<% end -%>
-<% end -%>
-<% else -%>
-PEERDNS=no
-PEERROUTES=no
-DEFROUTE=no
-<% end -%>
-<% if @host.ip6.present? -%>
-IPV6INIT=yes
-IPV6_AUTOCONF=no
-IPV6ADDR=<%= @host.ip6 %>
-<% if subnet6 && subnet6.gateway.present? -%>
-IPV6_DEFAULTGW=<%= subnet6.gateway %>
-<% end -%>
-IPV6_PEERDNS=no
-<% end -%>
-<% if vlan -%>
-VLAN=yes
-<% elsif alias_type -%>
-TYPE=Alias
-<% end -%>
-EOF
-
-<% end %>
+snippet: true
+model: ProvisioningTemplate
+kind: snippet
+-%>
+<%- bonded_interfaces = [] -%>
+<%- @host.bond_interfaces.each do |bond| -%>
+<%=   "# #{bond.identifier} interface" %>
+<%=   snippet('kickstart_ifcfg_get_identifier_names', :variables => { :identifier => bond.identifier }) %>
+<%-   ifcfg = snippet('kickstart_ifcfg_bond_interface', :variables => {
+                        :interface => bond,
+                        :subnet => bond.subnet,
+                        :subnet6 => bond.subnet6,
+                        :dhcp => bond.subnet.nil? ? false : bond.subnet.dhcp_boot_mode? }) -%>
+<%=   save_to_file('/etc/sysconfig/network-scripts/ifcfg-$sanitized_real', ifcfg) %>
```

Comment:
any suggestion how to make it better? would a comment that it's set by the snippet kickstart_ifcfg_get_identifier_names help? we can't get the value in ERB since it's only known when the template is executed on the target